esdt-safe/src/lib.rs

```diff
@@ -133,7 +131,9 @@ pub trait EsdtSafe {
         // used prevent transaction spam
         self.reserve_fee(&caller);
 
+        let sender_nonce = self.transactions_by_nonce(&caller).len() + 1;
```

Comment:
why +1 ?

Reply:
Previously, the nonce was taken AFTER pushing into the list, now it's taken BEFORE, hence the +1.